airflow/providers/google/cloud/operators/dataplex.py

```diff
@@ -610,3 +612,740 @@ def execute(self, context: Context) -> None:
+        account from the list granting this role to the originating account (templated).
+    :param fail_on_dq_failure: If set to true and not all Data Quality scan rules have been passed,
+        an exception is thrown. If set to false and not all Data Quality scan rules have been passed,
+        execution will finish with success.
+    :param wait_for_result: Wait for job to finish.
+    :param run_timeout: The amount of time, in seconds, to wait for the result.
+
+    :return: Dict representing DataScanJob.
+        When the job completes with a successful status, information about the Data Quality result
+        is available.
+    """
+
+    template_fields = ("project_id", "data_scan_id", "impersonation_chain")
+
+    def __init__(
+        self,
+        project_id: str,
+        region: str,
+        data_scan_id: str,
+        job_id: str | None = None,
+        api_version: str = "v1",
+        retry: Retry | _MethodDefault = DEFAULT,
+        timeout: float | None = None,
+        metadata: Sequence[tuple[str, str]] = (),
+        gcp_conn_id: str = "google_cloud_default",
+        impersonation_chain: str | Sequence[str] | None = None,
+        fail_on_dq_failure: bool = False,
+        wait_for_results: bool = True,
+        fail_on_job_failure: bool = False,
+        run_timeout: float | None = None,
+        *args,
+        **kwargs,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.project_id = project_id
+        self.region = region
+        self.data_scan_id = data_scan_id
+        self.job_id = job_id
+        self.api_version = api_version
+        self.retry = retry
+        self.timeout = timeout
+        self.metadata = metadata
+        self.gcp_conn_id = gcp_conn_id
+        self.impersonation_chain = impersonation_chain
+        self.fail_on_dq_failure = fail_on_dq_failure
+        self.fail_on_job_failure = fail_on_job_failure
+        self.wait_for_results = wait_for_results
+        self.run_timeout = run_timeout
+
+    def execute(self, context: Context) -> dict:
+        hook = DataplexHook(
+            gcp_conn_id=self.gcp_conn_id,
+            api_version=self.api_version,
+            impersonation_chain=self.impersonation_chain,
+        )
+        # fetch the last job
+        if not self.job_id:
+            jobs = hook.list_data_scan_jobs(
+                project_id=self.project_id,
+                region=self.region,
+                data_scan_id=self.data_scan_id,
+                retry=self.retry,
+                timeout=self.timeout,
+                metadata=self.metadata,
+            )
+            job_ids = [DataScanJob.to_dict(job) for job in jobs]
+            if not job_ids:
+                raise AirflowException("There are no jobs, you should create one before.")
+            job_id = job_ids[0]["name"]
+            self.job_id = job_id.split("/")[-1]
+
+        if self.wait_for_results:
```

Comment:
i can't see the implementation of waiting logic here:
If no results available wait until timeout for the results to be available

Reply:
wait logic is available within
`wait_for_data_scan_job`
```
if self.wait_for_results:
            hook.wait_for_data_scan_job(
```